crates/cast/bin/tx.rs

```diff
@@ -270,7 +271,29 @@ where
         }
 
         if self.tx.gas.is_none() {
-            self.tx.gas = Some(self.provider.estimate_gas(&self.tx).await?);
+            let resp = self.provider.estimate_gas(&self.tx).await;
+            // println!("{:?}", resp);
+            match resp  {
+                Ok(gas_estimate) => {
+                    self.tx.gas = Some(gas_estimate);
+                },
+                Err(e) => {
+                    match e {
+                        RpcError::ErrorResp(error_payload) => {
+                            let err_str = format!(
+                                "{} with data: {}",
+                                error_payload,
+                                error_payload.data.as_ref()
+                                    .map_or_else(String::new, |err_data| err_data.get().trim_matches('"').to_string())
+                            );
+                            return Err(ErrReport::msg(err_str))
+                        }
+                        _ =>{
+                            return Err(e.into())
+                        }
+                    };
```

Comment:
I think we try using

`error_payload.as_decoded_error::<GenericContractError>()` here?

Reply:
I try, but it seems error_payload can not decode as GenericContractError.

Reply:
```  
          Err(e) => {
                match e {
                    RpcError::ErrorResp(error_payload) => {
                        return Err(match error_payload.as_decoded_error::<GenericContractError>(false) {
                            Some(contract_error) => ErrReport::msg(contract_error.to_string()),
                            None => ErrReport::msg("No contract error"),
                        });
                    }
                    _ => {
                        return Err(e.into())
                    }
                };
            }
```

result in No contract error

Reply:
It seems needs the abi.